### PR TITLE
Ravil/graphs

### PR DESCRIPTION
**Pull request by ravil-mobile**
Added  CUDA/HIP graphs (using graphs capturing feature)

Depends on the following PR: https://github.com/SeisSol/Device/pull/20

### CODE REVIEW

File: preprocessing/science/rconv/CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +24,7 @@ if (NOT PROJ4_FOUND)
         pkg_check_modules(PROJ4 QUIET proj>=4.9)
         if (PROJ4_FOUND)
             set(CMAKE_INSTALL_RPATH "${PROJ4_LIBRARY_DIRS};${CMAKE_INSTALL_RPATH}")
+            set(PROJ4_LIBRARIES "${PROJ4_LINK_LIBRARIES}")
```
Comment (krenzland) [inline preprocessing/science/rconv/CMakeLists.txt:27]: This is a bit random - was this tested or just committed by accident?

File: src/Initializer/BasicTypedefs.hpp [hunk 2]
```diff
@@ -65,4 +65,13 @@ struct PlasticityData {
   real mufactor;
 };
 
+enum ComputeGraphType {
```
Comment (krenzland) [inline src/Initializer/BasicTypedefs.hpp:68]: should be enum class

**Review by krenzland — CHANGES_REQUESTED** [review]
Looks mostly good!

**Comment by ravil-mobile** [comment]
Hi @krenzland, Thanks for reviewing the code!

File: preprocessing/science/rconv/CMakeLists.txt (hunk 1 above)
Comment (ravil-mobile) [reply preprocessing/science/rconv/CMakeLists.txt:27]: Hi, yes. It was tested. There is a difference how pkg-config and cmake name LINK_LIBRARIES. I had to fix it for the Docker CI image.

File: src/Initializer/BasicTypedefs.hpp (hunk 2 above)
Comment (ravil-mobile) [reply src/Initializer/BasicTypedefs.hpp]: done

**Review by krenzland — APPROVED** [review]

**Pull request merged**